hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -773,34 +773,28 @@ public boolean merge(MapMergeTypes mergingEntry, SplitBrainMergePolicy<Data, Map
             storage.put(key, record);
             eventJournal.writeUpdateEvent(mapContainer.getEventJournalConfig(), mapContainer.getObjectNamespace(), partitionId,
                     key, null, record.getValue());
-        } else {
-            oldValue = record.getValue();
-            MapMergeTypes existingEntry = createMergingEntry(serializationService, record);
-            newValue = mergePolicy.merge(mergingEntry, existingEntry);
-            // existing entry will be removed
-            if (newValue == null) {
-                removeIndex(record);
-                mapDataStore.remove(key, now);
-                onStore(record);
-                eventJournal.writeUpdateEvent(mapContainer.getEventJournalConfig(), mapContainer.getObjectNamespace(),
-                        partitionId, key, oldValue, null);
-                storage.removeRecord(record);
-                return true;
-            }
-            if (newValue == mergingEntry.getValue()) {
-                mergeRecordExpiration(record, mergingEntry);
-            }
-            // same with the existing entry so no need to map-store etc operations.
-            if (recordComparator.isEqual(newValue, oldValue)) {
-                return true;
-            }
-            newValue = mapDataStore.add(key, newValue, now);
+            saveIndex(record, null);
+            return newValue != null;
```

Comment:
I believe this can be `return true;` since we do the null-check above.

Reply:
I checked the `MapDataStore` implementations, and it looks that they never return `null`. So this should be safe to do, yes.